lib/Parser/Parse.cpp

```diff
@@ -5814,52 +5796,53 @@ void Parser::ParseTopLevelDeferredFunc(ParseNodeFnc * pnodeFnc, ParseNodeFnc * p
 
     m_ppnodeVar = &pnodeFnc->pnodeVars;
 
-    if (pnodeFncParent != nullptr
-        && m_currDeferredStub != nullptr
-        // We don't create stubs for function bodies in parameter scope.
-        && pnodeFnc->pnodeScopes->blockType != PnodeBlockType::Parameter)
```

Comment:
This probably isn't a big deal, since as you say these expression tend to be small, but what's the issue with the text extents for such a function?

Reply:
The text extents for these lambdas are computed specially in `ParseExpressionLambdaBody`. I think it boils down to concise-body lambdas not having a closing brace character which means the token we scan to at the end of the body is not part itself of the body. We usually use the ichMin character of this token to find the end of the function body - since it's usually a brace, this usually gives us the correct extents of the body even if there's lots of whitespace. For concise-body lambda we use the ichLim of the previous token instead.